tests/std/tests/P0896R4_ranges_alg_find_first_of/test.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+//
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::find_first_of, ranges::iterator_t, std::array, std::same_as, std::to_address;
+    using P = std::pair<int, int>;
+
+    // Validate dangling story
+    STATIC_ASSERT(same_as<decltype(find_first_of(borrowed<false>{}, array<int, 42>{})), ranges::dangling>);
+    STATIC_ASSERT(same_as<decltype(find_first_of(borrowed<true>{}, array<int, 42>{})), int*>);
+
+    const array pairs = {P{0, 42}, P{1, 42}, P{2, 42}, P{3, 42}, P{4, 42}, P{5, 42}, P{6, 42}};
+
+    const auto pred = [](const int x, const int y) { return x == y + 1; };
+
+    const array good_needle = {29, 1};
+    {
+        // Validate range overload [found case]
+        const auto result = find_first_of(move_only_range{pairs}, good_needle, pred, get_first);
+        STATIC_ASSERT(same_as<decltype(result), const iterator_t<move_only_range<const P>>>);
+        assert(&*result.base() == &pairs.begin()[2]);

```

Comment:
Style: Should you consistently use `to_address(result.base())`? Occurs below. Feel free to disregard.

Reply:
Yes, I think these should consistently `to_address(result.base()) == pairs.data() + meow`.